Source/Turbo Navigator/TurboNavigator.swift

```diff
@@ -113,7 +113,7 @@ extension TurboNavigator: SessionDelegate {
 
     public func sessionDidFinishFormSubmission(_ session: Session) {
         if session == modalSession {
-            self.session.clearSnapshotCache()
+            self.session.perform(#selector(Session.clearSnapshotCache), with: nil, afterDelay: 1)
```

Comment:
If you moved this logic to `sessionDidStartFormSubmission()` instead without the delay, do you see the expected result? That might be an ok approach.

Reply:
No, that doesn't seem to do the trick. From what I can gather, we need to _delay_ the call to `clearSnapshotCache()`, not move it up.

Reply:
>  I'm not able to nail down why Turbo.session.clearCache() seems to silently fail in this scenario but I'm guessing it has something to do with the view not being 100% "visible". As in, the modal might be the active web view so this JavaScript call might not fully execute.

`clearCache()` is indeed called before the modal is dismissed. If your thesis holds true, then we need to somehow delay clearing the cache until after the modal view has been dismissed. I'll think about it.

Reply:
I tested the demo app thoroughly and only once hit a javascript error when calling `window.turboNative.clearSnapshotCache`. I thought I'd hit it again, but no. I didn't copy the error, but it was along the lines of "couldn't execute in this document."

I was able to find a solution though. Whether it's elegant is up for debate :)
The idea is to set a flag on `Session` that a cache clear is needed and check the flag on `visitableViewWillAppear`. The flag is set in `TurboNavigator` in `sessionDidFinishFormSubmission`.

```
public class Session: NSObject {
    ...
    var needsCacheClearing: Bool = false
    ...
}
```

```
 public func visitableViewWillAppear(_ visitable: Visitable) {
        if needsCacheClearing {
            clearSnapshotCache()
        }
        ...
}
```

```
public func sessionDidFinishFormSubmission(_ session: Session) {
	if session == modalSession {
	    self.session.needsCacheClearing = true
	}
}
```

Reply:
Sure thing, I can take a look this week.

While only slightly tangential, it's starting to feel like #150 (and subsequently #174) will make sense to merge.

Reply:
This works great! @svara, I opened PR #182 with your changes. Can you please confirm I put the `if needsCacheClearing...` in the right spot?

Reply:
@joemasilotti The PR looks great. @jayohms The current flag name is still `needsCacheClearing`. I don't feel strongly about the name, but I agree that `snapshotCacheIsStale` may be a better name.

Reply:
@svara have you seen this? https://github.com/hotwired/turbo-ios/pull/181

It's a fairly similar issue - but the opposite problem. Knowing when to take a snapshot cache instead of deleting it. We should use consistent language and implementation if possible. Can you take a look at that one?

Reply:
@jayohms I've started reviewing the PR. The solution looks solid, but I need to spend some more time familiarizing myself with caching in general.